core/models.py

```diff
@@ -179,6 +179,8 @@ class AlgorithmResult(models.Model):
     created = models.DateTimeField(default=timezone.now)
     data = models.FileField(upload_to='results')
     log = models.FileField(upload_to='results_logs', null=True, blank=True)
+    log_message = models.TextField(default='This only show the end of the log')
```

Comment:
We don't actually want to add these to the models as fields, since that adds them to the database.  Similarly, we don't want the task to generate this.  The task just saves the log file.

Rather, there should just be a function in the `AdminModel` for `AlgorithmResultAdmin` and `ScoreResultAdmin` (much like what is currently in master, but generalized for the two models as a common function).

I'd expect that there shouldn't need to be a migration for this, as the core models shouldn't change.

Reply:
So both log_message and log_preview should not be added to the models?
Instead, they should be a generalized function that takes the file and return corresponding outputs?

Reply:
Neither should be added to the models.

For logs, we just want to show the end of the log (your log_message).

If the *data* is known to be a text mime-type, then we will want to show the beginning of the data.  Currently, we don't check or store the data mime type; we, ideally, would have different previews based on the type.  For, now if we showed the beginning of the data as text, that is still useful for testing.

Reply:
For log_message, it should be a message telling the user that we are only showing the end of the log or it should show the end of the log file(the last line of the log file)?
And for log_preview, show the beginning of the data as text?

Reply:
For logs, we want to show the last ~100kB of the log file as text.  If it isn't the complete file, we should mention that.  For instance, Circle CI does this by printing:
```
Your output is too large to display in the browser.
Only the last 400000 characters are displayed.
```
For our use case this should probably be:
```
The log output is too large to display in the browser.
Only the last 100000 characters are displayed.
```
where that 100000 should be a variable in the function so we aren't duplicating it in multiple places.

For generality, we probably can have a single utility function, `def _text_preview(filefield, maxlen=100000, showEnd=True):`, where for logs it would call it with `filefield=<model>.log`, and for data it would call it with `filefield=<model>.data, showEnd=False)`.